include/net/mqtt_sn.h

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2022 René Beckmann
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file mqtt_sn.h
+ *
+ * @defgroup mqtt_sn_socket MQTT Client library
+ * @ingroup networking
+ * @{
+ * @brief MQTT-SN Client Implementation
+ *
+ * @details
+ * MQTT-SN Client's Application interface is defined in this header.
+ * Targets protocol version 1.2.
+ *
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_MQTT_SN_H_
+#define ZEPHYR_INCLUDE_NET_MQTT_SN_H_
+
+#include <stddef.h>
+
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <net/buf.h>
+#include <net/net_ip.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Quality of Service. QoS 0-2 work the same as basic MQTT, QoS -1 is an MQTT-SN addition.
+ */
+enum __attribute__((packed)) mqtt_sn_qos {
```

Comment:
Why do they need to be packed?

Reply:
And if it really needs to be packed, there is the `__packed` define that can be used here.

Reply:
It does make the parameter-union in the encoder and decoder parts much smaller. In an earlier version I had to store those for re-sends. Now, they are still used in the `mqtt_sn_publish` and `mqtt_sn_topic` structs. I thought using 3 bytes instead of 12 might be useful (lots of assumptions here).

Reply:
removed these